daft/series.py

```diff
@@ -18,6 +18,12 @@ class Series:
     def __init__(self) -> None:
         raise NotImplementedError("We do not support creating a Series via __init__ ")
 
+    def __iter__(self) -> Iterator[Any]:
+        """Return an iterator over the elements of the Series."""
+        arrow_data = self.to_arrow()
+        for item in arrow_data:
+            yield None if item is None else item.as_py()
```

Comment:
This is great. Maybe we can have an issue (or do it in this PR) to update docs/examples to use this.

Reply:
Yeah totally! I've been going through docs replacing examples. But I've also been trying to measure memory usage in each example and we don't always get a win, so I'm digging a little deeper before we land this